runtimes/spiritnet/src/lib.rs

```diff
@@ -784,15 +784,23 @@ construct_runtime! {
 
 impl did::DeriveDidCallAuthorizationVerificationKeyRelationship for Call {
 	fn derive_verification_key_relationship(&self) -> Option<did::DidVerificationKeyRelationship> {
+		fn single_key_relationship(calls: &[Call]) -> Option<did::DidVerificationKeyRelationship> {
```

Comment:
This compiles for me. I am quite sure the `as_ref()` is not needed, while I am not super sure why you went with a `flat_map()` instead of a simple `map()`? When we iterate, we are not iterating over options, so we don't need to flatten the map, as we need in the beginning because of the return type of `get()`.

```suggest
		fn single_key_relationship(calls: &[Call]) -> Option<did::DidVerificationKeyRelationship> {
			let init = calls.get(0).map(Call::derive_verification_key_relationship).flatten()?;
			calls
				.iter()
				.map(Call::derive_verification_key_relationship)
				.try_fold(init, |acc, next| if next == acc { Some(acc) } else { None })
		}
```

Reply:
M: Didn't we also agree to change this to a `Result<did::DidVerificationKeyRelationship, DispatchError>`?

Reply:
yeah wasn't sure if it only was a "this would be cool" thing.
The issue is that we would need to define the error in the DID pallet. The DID pallet doesn't know which errors we want to return here. We could try to keep the error generic?

* `NotADidCall` for calls that don't support the did origin
* `InvalidCallArgument` for calls with arguments that are not allowed in the did call

Reply:
DispatchErrors have to be defined in a Pallet. So we have the issue again that the pallet logic depends on the runtime. :/